packages/grid/src/vaadin-grid.js

```diff
@@ -546,6 +546,18 @@ class Grid extends ElementMixin(
       col.width = 'auto';
       col._origFlexGrow = col.flexGrow;
       col.flexGrow = 0;
+
```

Comment:
There's already quite a lot of `auto-width` related logic in the main file `vaadin-grid.js`. For readability, I'd suggest extracting all the related logic to `vaadin-grid-auto-width-mixin.js` to avoid further bloating the main file. This can also be done separately if you prefer keeping this PR minimal.